src/routes/webhook/webhook-receiver-post.ts

```diff
@@ -0,0 +1,77 @@
+import { BinaryLike, createHmac } from "crypto";
+import { Request, Response } from "express";
+import { validationResult } from "express-validator";
+import { getLogger } from "~/src/config/logger";
+import { GitHubServerApp } from "~/src/models/git-hub-server-app";
+import { WebhookContext } from "./webhook-context";
+
+export const WebhookReceiverPost = async (request: Request, response: Response): Promise<void> => {
+	const errors = validationResult(request);
+	if (!errors.isEmpty()) {
+		response.status(400).json({ errors: errors.array() });
+		return;
+	}
+	const logger = getLogger("webhook.receiver");
+	const eventName = request.headers["x-github-event"] as string;
+	const signatureSHA256 = request.headers["x-hub-signature-256"] as string;
+	const id = request.headers["x-github-delivery"] as string;
+	const uuid = request.params.uuid;
+	let webhookSecret: string = process.env.WEBHOOK_SECRET!;
+	const payload = request.body;
+	try {
+		if (uuid != "cloud") {
+			const gitHubServerApp = await GitHubServerApp.findForUuid(uuid);
+			if (!gitHubServerApp) {
+				response.status(400).send("GitHub app not found");
+				return;
+			}
+			webhookSecret = gitHubServerApp?.webhookSecret;
+		}
+		const verification = createHash(JSON.stringify(payload), webhookSecret);
+		if (verification != signatureSHA256) {
+			response.status(400).send("signature does not match event payload and secret");
+			return;
+		}
+
+		const webhookContext = new WebhookContext({
+			id: id,
+			name: eventName,
+			payload: payload,
+			log: logger
+		});
+
+		const action = "action" in payload ? payload.action : null;
```

Comment:
I don't understand the point of this.  It's the same as doing `const action = payload.action`.  We use `undefined` instead of `null` wherever possible